x/interchainqueries/keeper/keeper.go

```diff
@@ -260,3 +263,26 @@ func (k Keeper) checkRegisteredQueryExists(ctx sdk.Context, id uint64) bool {
 
 	return store.Has(types.GetRegisteredQueryByIDKey(id))
 }
+
+func (k Keeper) CollectDeposit(ctx sdk.Context, queryInfo types.RegisteredQuery) error {
+	owner, err := queryInfo.GetOwnerAddress()
+	if err != nil {
+		panic(err.Error())
+	}
+
+	err = k.bank.SendCoinsFromAccountToModule(ctx, owner, types.ModuleName, sdk.NewCoins(queryInfo.GetDepositCoin()))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (k Keeper) MustPayOutDeposit(ctx sdk.Context, deposit sdk.Coin, sender sdk.AccAddress) error {
+	err := k.bank.SendCoinsFromModuleToAccount(ctx, types.ModuleName, sender, sdk.NewCoins(deposit))
+	if err != nil {
+		panic(types.ErrCannotPayDeposit.Error())
```

Comment:
The same about `panic()`

Reply:
But it means that something wrong with code. If the module cannot pay, it means the escrow account has failed. This error is much more serious: an invariant has been violated and the whole application must be terminated.

Reply:
Agree. But please include `err` into the panic call to make error trace easier to understand what went exactly